yarn-project/world-state/src/native/native_world_state_integration.test.ts

```diff
@@ -0,0 +1,292 @@
+      expect(native.lowLeavesWitnessData).toEqual(js.lowLeavesWitnessData);
+
+      await assertSameTree(treeId, true);
+    });
+
+    it.each<[string, FrTreeId, Fr[]]>([
+      [MerkleTreeId[MerkleTreeId.NOTE_HASH_TREE], MerkleTreeId.NOTE_HASH_TREE, Array(64).fill(0).map(Fr.random)],
+      [MerkleTreeId[MerkleTreeId.L1_TO_L2_MESSAGE_TREE], MerkleTreeId.NOTE_HASH_TREE, Array(64).fill(0).map(Fr.random)],
+    ])('inserting real leaves into %s', async (_, treeId, leaves) => {
+      await Promise.all([nativeWS.appendLeaves(treeId, leaves), legacyWS.appendLeaves(treeId, leaves)]);
+
+      await assertSameTree(treeId, true);
+    });
+
+    it.each<[string, MerkleTreeId, number[]]>([
+      [MerkleTreeId[MerkleTreeId.NULLIFIER_TREE], MerkleTreeId.NULLIFIER_TREE, [0, 1, 10, 127, 128, 256]],
+      [MerkleTreeId[MerkleTreeId.NOTE_HASH_TREE], MerkleTreeId.NOTE_HASH_TREE, [0, 1, 10, 127, 128, 256]],
+      [MerkleTreeId[MerkleTreeId.PUBLIC_DATA_TREE], MerkleTreeId.PUBLIC_DATA_TREE, [0, 1, 10, 127, 128, 256]],
+      [MerkleTreeId[MerkleTreeId.L1_TO_L2_MESSAGE_TREE], MerkleTreeId.NOTE_HASH_TREE, [0, 1, 10, 127, 128, 256]],
+    ])('sibling paths to initial leaves match', async (_, treeId, leafIndices) => {
+      for (const index of leafIndices) {
+        const [nativeSB, legacySB] = await Promise.all([
+          nativeWS.getSiblingPath(treeId, BigInt(index), true),
+          legacyWS.getSiblingPath(treeId, BigInt(index), true),
+        ]);
+        expect(nativeSB.toFields()).toEqual(legacySB.toFields());
+      }
+    });
+
+    it.each<[string, IndexedTreeId, Buffer, (b: Buffer) => bigint]>([
+      [
+        MerkleTreeId[MerkleTreeId.NULLIFIER_TREE],
+        MerkleTreeId.NULLIFIER_TREE,
+        new NullifierLeaf(Fr.random()).toBuffer(),
+        b => NullifierLeaf.fromBuffer(b).getKey(),
+      ],
+      [
+        MerkleTreeId[MerkleTreeId.PUBLIC_DATA_TREE],
+        MerkleTreeId.PUBLIC_DATA_TREE,
+        new PublicDataTreeLeaf(Fr.random(), Fr.random()).toBuffer(),
+        b => PublicDataTreeLeaf.fromBuffer(b).getKey(),
+      ],
+    ])('inserting real leaves into %s', async (_, treeId, leaf, getKey) => {
+      await Promise.all([nativeWS.batchInsert(treeId, [leaf], 0), legacyWS.batchInsert(treeId, [leaf], 0)]);
+
+      const [nativeLeafIndex, legacyLeafIndex] = await Promise.all([
+        nativeWS.getPreviousValueIndex(treeId, getKey(leaf), true),
+        legacyWS.getPreviousValueIndex(treeId, getKey(leaf), true),
+      ]);
+      expect(nativeLeafIndex).toEqual(legacyLeafIndex);
+
+      const [nativeSB, legacySB] = await Promise.all([
+        nativeWS.getSiblingPath(treeId, nativeLeafIndex!.index, true),
+        legacyWS.getSiblingPath(treeId, legacyLeafIndex!.index, true),
+      ]);
+
+      expect(nativeSB.toFields()).toEqual(legacySB.toFields());
+    });
+  });
+
+  describe('Block synch', () => {
+    it.skip('syncs a new block from empty state', async () => {
```

Comment:
Need to investigate why this does not work as expected. The only mismatch is the nullifier tree root. I think the the world state cpp code might be wrong so I've disabled this while I investigate further